add necessary models and methods for the controllers to be built

STOzaki opened the pull request:
RestTemplate will not throw an error if it receives http 400 or 500 series. All of the other PRs revolve around this PR.

STOzaki committed:
add RestTemplate to application and error handling

RestTemplate will not throw an error if it receives http 400 or 500 series

kingjon3377 commented on src/main/java/com/st/utopia/agent/util/RestTemplateResponseErrorHandler.java:
```diff
@@ -0,0 +1,17 @@
+public class RestTemplateResponseErrorHandler implements ResponseErrorHandler {
+	@Override
+	public boolean hasError(ClientHttpResponse httpResponse) {
+		/* By default it would return true and throw a error if it was anything other
```
I'd suggest making this a comment on the method, not merely the return value.

STOzaki replied:
Will do.

STOzaki committed:
add comment in restTemplate on the method instead of in the method

STOzaki committed:
add model for controllers to use

These models will be used by the booking and cancellation services

STOzaki committed:
commented empty method

codacy states that empty methods should be commented otherwise no one knows if that was intentional or not
